solcore/solar_cell_solver.py

```diff
@@ -67,7 +68,13 @@ def solar_cell_solver(solar_cell, task, user_options=None):
     options.T = solar_cell.T
 
     if task == 'optics':
-        solve_optics(solar_cell, options)
+        calculated = hasattr(solar_cell[0], 'absorbed')
```

Comment:
Adding this option to avoid recalculating the optics unless requested is excellent. However, I think it should be inside the `solve_optics` function below. As it is now, this check is also done in `solve_iv`, `solve_qe` and `solve_short_circuit`, which adds redundant code. Within `solve_optics` it only needs to be written once.

Reply:
Yes, good point. I will change this.